ipa-core/src/protocol/ipa_prf/malicious_security/verifier.rs

```diff
@@ -10,6 +10,43 @@ use crate::{
     },
 };
 
+/// This function computes the shares that sum to zero from the zero-knowledge proofs.
+pub fn compute_g_differences<F, const P: usize, const λ: usize>(
+    zkp: &Vec<[F; P]>,
+    challenges: &[F],
+) -> Vec<F>
+where
+    F: PrimeField,
+{
+    // compute denominator
+    let lagrange_denominator: CanonicalLagrangeDenominator<F, P> =
+        CanonicalLagrangeDenominator::<F, P>::new();
+
+    // compute g_r with "empty" first spot
+    let g_r = iter::once(F::ZERO)
```

Comment:
I think we should instead initialize this to -m/2. In the protocol, they suggest P_{i+1} initialize to -m/2 and P_{i-1} initialize to 0.

By doing it here, we avoid the need for an ugly modification after calling this function.